4.	Frequency distributions, Averages, Variability

1. Frequency Distributions

A frequency distribution shows how often each value appears in a dataset.

Example: Frequency Distribution of Wine Quality Ratings

In [1]:
import pandas as pd

# Sample data: Wine quality ratings
data = {'Quality': [5, 6, 5, 7, 8, 6, 5, 7, 6, 8, 9, 4]}
df = pd.DataFrame(data)

# Frequency distribution
frequency_distribution = df['Quality'].value_counts().sort_index()
print(frequency_distribution)


Quality
4    1
5    3
6    3
7    2
8    2
9    1
Name: count, dtype: int64


Interpretation:

    Quality rating 5 appears 3 times.
    Quality rating 6 appears 3 times.
    Quality rating 4 and 9 appear once each.

2. Averages (Measures of Central Tendency)

Averages summarize data with a single representative value. The most common are:

    Mean: The arithmetic average.
    Median: The middle value when data is sorted.
    Mode: The most frequently occurring value.

In [2]:
# Sample data: Alcohol content
data = {'Alcohol': [9.5, 10.1, 9.8, 10.5, 9.7, 10.3, 9.9, 10.2, 9.6, 10.0]}
df = pd.DataFrame(data)

# Mean
mean_alcohol = df['Alcohol'].mean()

# Median
median_alcohol = df['Alcohol'].median()

# Mode
mode_alcohol = df['Alcohol'].mode()[0]

print(f"Mean Alcohol: {mean_alcohol}")
print(f"Median Alcohol: {median_alcohol}")
print(f"Mode Alcohol: {mode_alcohol}")


Mean Alcohol: 9.959999999999999
Median Alcohol: 9.95
Mode Alcohol: 9.5


Interpretation:

    The average alcohol content is 9.96%.
    The median is 9.95%, indicating the middle point of the data.
    The mode is 9.5%, the most frequently occurring value.

3. Variability (Measures of Spread)

Variability measures how spread out the data is. Common measures include:

    Range: Difference between the maximum and minimum values.
    Variance: The average of squared differences from the mean.
    Standard Deviation: The square root of variance, indicating how much the values deviate from the mean.

In [3]:
# Sample data: Residual sugar content
data = {'Residual Sugar': [1.5, 2.1, 2.8, 1.9, 2.5, 1.7, 2.3, 2.0, 2.6, 1.8]}
df = pd.DataFrame(data)

# Range
range_sugar = df['Residual Sugar'].max() - df['Residual Sugar'].min()

# Variance
variance_sugar = df['Residual Sugar'].var()

# Standard Deviation
std_dev_sugar = df['Residual Sugar'].std()

print(f"Range of Residual Sugar: {range_sugar}")
print(f"Variance of Residual Sugar: {variance_sugar}")
print(f"Standard Deviation of Residual Sugar: {std_dev_sugar}")


Range of Residual Sugar: 1.2999999999999998
Variance of Residual Sugar: 0.17733333333333332
Standard Deviation of Residual Sugar: 0.42110964526276684


Interpretation:

    The range of residual sugar is 1.3 g/L.
    The variance shows how much the sugar content varies around the mean.
    The standard deviation of 0.407 indicates the average deviation from the mean sugar content.